common/modules/gcp-project/main.tf

```diff
@@ -21,9 +21,14 @@ locals  {
             }"
 }
 
+resource "random_string" "id" {
+  length      = 4
+  min_numeric = 4
+}
+
 resource "google_project" "project" {
   name            = "gpii-gcp-${var.project_name}"
-  project_id      = "gpii-gcp-${var.project_name}"
+  project_id      = "gpii-gcp-${var.project_name}-${random_string.id.result}"
```

Comment:
Can you elaborate on why we are generating a nonce both here and in `set_vars()`. Seems like one of these calculations is unnecessary. Am I missing something?